openfasoc/generators/gdsfactory-gen/relational.py

```diff
@@ -0,0 +1,754 @@
+        If found, return info, else return None"""
+        for param in self.params:
+            if param["name"]==param_name:
+                return param
+        return None
+    def __str__(self):
+        return str(self.params)
+    
+    def __construct_glayers(self) -> dict:
+        glayerids = ['met1', 'met2', 'met3', 'met4', 'met5', 'metal1', 'metal2', 'metal3', 'metal4', 'metal5', 'capmet', 'active', 'diffusion', 'tap', 'welltap', 'n+s/d', 'p+s/d', 'polysilicon', 'poly', 'p+', 'n+', 'deep nwell', 'pwell', 'nwell', 'dnwell']
+        glayerids += ["metal 1", "metal 2","metal 3","metal 4","metal 5","met 1","met 2","met 3","met 4","met 5"]
+        glayerids += ["via 1", "via 2","via 3","via 4","via1","via2","via3","via4","metal contact","mcon","metal con"]
+        glayers = dict()
+        for glayerid in glayerids:
+            if "p+" in glayerid:
+                glayers[glayerid] = "p+s/d"
+            elif "n+" in glayerid:
+                glayers[glayerid] = "n+s/d"
+            elif "poly" in glayerid:
+                glayers[glayerid] = "poly"
+            elif "cap" in glayerid:
+                glayers[glayerid] = "capmet"
+            elif "tap" in glayerid:
+                glayers[glayerid] = "active_tap"
+            elif "active" in glayerid or "diff" in glayerid:
+                glayers[glayerid] = "active_diff"
+            elif "well" in glayerid:
+                if "d" in glayerid:
+                    glayers[glayerid] = "dnwell"
+                elif "n" in glayerid:
+                    glayers[glayerid] = "nwell"
+                else:
+                    glayers[glayerid] = "pwell"
+            elif "via" in glayerid:
+                for i in range(1,5):
+                    if str(i) in glayerid:
+                        glayers[glayerid] = f"via{i}"
+            elif "met" in glayerid:
+                for i in range(1,6):
+                    if str(i) in glayerid:
+                        glayers[glayerid] = f"met{i}"
+            elif "con" in glayerid:
+                glayers[glayerid] = "mcon"
+            else:
+                raise ValueError("could not parse this id")
+        return glayers
+    def __cfgformat_from_list(self, strlist: list[str]):
+        rtrstr = str()
+        for ele in strlist:
+            rtrstr += "\'" + ele + "\' | "
+        rtrstr =rtrstr.removesuffix("| ")
+        return rtrstr
+    def __custom_tokenize(self, sentence: str) -> list:
+        sentence = sentence.strip().removesuffix(".").removeprefix("with")
+        # save and replace all instances of strings with indicator words
+        string_pattern = re.compile(r'\'[^\']*\'|\"[^\"]*\"')
+        string_list = [str(match.group()) for match in string_pattern.finditer(sentence)]
+        string_list = [w.replace("\"","").replace("'","").strip() for w in string_list]
+        sentence = re.sub(string_pattern, "sTRingindICatorWOrd", sentence)
+        # add a space after every symbol (except period and _)
```

Comment:
What would happen if there was already a space?

Reply:
I believe I would add another space. Should be fine, can you find a case where it break parsing? if so please write a test case for it and we can start building a regression tests

Reply:
If adding an extra space doesn't break it then I think it is fine.